### PR TITLE
DNS upstream servers report

### LINKED ISSUE #1571: Print out info on upstream servers for weaveDNS debugging
When remote troubleshooting it can be useful to know what weave thinks its upstream servers are.

### PR DESCRIPTION
**Pull request by inercia**
Fixes #1571.

### CODE REVIEW

**Comment by rade** [comment]
quick comment. the new field should come earlier everywhere - it's part of the configuration rather than dynamic state, and the former should (generally) precede the latter.

**Comment by inercia** [comment]
> quick comment. the new field should come earlier everywhere - it's part of the configuration rather than dynamic state, and the former should (generally) precede the latter.

You mean `Status.Upstream`?

File: prog/weaver/http.go [hunk 1]
```diff
@@ -25,6 +27,12 @@ var rootTemplate = template.New("root").Funcs(map[string]interface{}{
 		}
 		return count
 	},
+	"upstreamServers": func(servers []string) string {
+		if len(servers) == 0 {
+			return "WARNING: no servers !!"
```
Comment (inercia) [inline prog/weaver/http.go]: Any suggestions on how we should warn users when no servers are configured?

**Comment by rade** [comment]
> You mean `Status.Upstream`?

Yes, but also in the template.

**Comment by inercia** [comment]
> > You mean `Status.Upstream`?

> Yes, but also in the template.

Ok. What do you think about this?:

```
       Service: dns
        Domain: weave.local.
      Upstream: 8.8.8.8, 8.8.4.4
           TTL: 1
       Entries: 0
```

**Comment by rade** [comment]
> What do you think about this?

Looks fine.

**Comment by inercia** [comment]
CircleCI fails on `640_proxy_restart_reattaches_test`... Rebuilding until it passes...

**Comment by awh** [comment]
Docs need updating to reflect the new output

**Comment by awh** [comment]
> Any suggestions on how we should warn users when no servers are configured?

It should just say

```
    Upstream: disabled
```

for consistency with the other fields.

**Comment by rade** [comment]
> `Upstream: disabled`

No. To me `disabled` means the user has taken specific action to disable something.

**Comment by inercia** [comment]
>> Upstream: disabled

> No. To me disabled means the user has taken specific action to disable something.

_"Unavailable"_ ? _"Not configured"_ ?

**Comment by rade** [comment]
neither. "unavailable" implies it's something that might fix it self. "Not configured" implies it's something the user has to configure (which generally they don't). 'none' perhaps, but I defer to @awh.

**Comment by awh** [comment]
I would go with `none`.

File: nameserver/status.go [hunk 2]
```diff
@@ -32,12 +33,15 @@ func NewStatus(ns *Nameserver, dnsServer *DNSServer) *Status {
...
 	}
 
 	return &Status{
 		dnsServer.domain,
+		dnsServer.upstream.Servers,
 		dnsServer.address,
 		dnsServer.ttl,
-		entryStatusSlice}
+		entryStatusSlice,
+	}
```
Comment (awh) [inline nameserver/status.go]: Did gofmt do this?

Comment (inercia) [inline nameserver/status.go]: I did it. I remember reading about never ending lines with `}` in Go because it makes `diffs` worse...

Comment (awh) [inline nameserver/status.go]: Sorry to be a pain, but can you undo it? The entire rest of the status code is written in this style, and *if* we're going to change it we should do so uniformly in a separate PR...

Comment (inercia) [inline nameserver/status.go]: Ok, I'll remove these changes (including the `import`s in `prog/weaver/http.go`)

Comment (inercia) [inline nameserver/status.go]: Done
